pkg/apis/migration/migration.go

```diff
@@ -407,3 +415,129 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
+				return
+			}
+			log.Debugf("QueryVolumeInfo successfully returned with result:  %v:", spew.Sdump(queryAllResult))
+			cnsVolumesMap := make(map[string]bool)
+			for _, vol := range queryAllResult.Volumes {
+				cnsVolumesMap[vol.VolumeId.Id] = true
+			}
+			log.Debugf("cnsVolumesMap:  %v:", cnsVolumesMap)
+			var resourcesToBeDeleted []string
+			for _, volumeMigrationResource := range volumeMigrationResourceList.Items {
+				if _, existsInCNSVolumesMap := cnsVolumesMap[volumeMigrationResource.Name]; !existsInCNSVolumesMap {
+					log.Debugf("Volume with id %s is not found in CNS", volumeMigrationResource.Name)
+					resourcesToBeDeleted = append(resourcesToBeDeleted, volumeMigrationResource.Name)
+				}
+			}
+			for _, volID := range resourcesToBeDeleted {
+				err = volumeMigrationInstance.DeleteVolumeInfo(ctx, volID)
+				if err != nil {
+					log.Warnf("failed to delete volume mapping CR for %s with error %+v", volID, err)
+					continue
+				}
+			}
+			log.Infof("Completed CnsVSphereVolumeMigration cleanup")
+			for i := 1; i <= volumeMigrationInstance.cnsConfig.Global.VolumeMigrationCRCleanupIntervalInHours; i++ {
+				time.Sleep(time.Duration(1 * time.Hour))
+			}
+		}
+	}()
+}
+
+// watcher watches on the vsphere.conf file mounted as secret
+func (volumeMigration *volumeMigration) watcher(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	cfgPath := common.GetConfigPath(ctx)
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Errorf("Failed to create fsnotify watcher. Err: %+v", err)
+		return err
+	}
+	go func() {
+		for {
+			log.Debugf("Waiting for event on fsnotify watcher")
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				log.Debugf("fsnotify event: %q", event.String())
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					log.Infof("Reloading Configuration")
+					err := volumeMigrationInstance.reloadConfiguration(ctx)
+					if err != nil {
+						log.Infof("Failed to reload configuration from: %q. Err: %+v", cfgPath, err)
+					} else {
+						log.Infof("Successfully reloaded configuration from: %q", cfgPath)
+					}
+				}
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					log.Errorf("fsnotify error: %+v", err)
+					return
+				}
+			}
+			log.Debugf("fsnotify event processed")
+		}
+	}()
+	cfgDirPath := filepath.Dir(cfgPath)
+	log.Infof("Adding watch on path: %q", cfgDirPath)
+	err = watcher.Add(cfgDirPath)
+	if err != nil {
+		log.Errorf("Failed to watch on path: %q. err=%v", cfgDirPath, err)
+	}
+	return err
+}
+
+//reloadConfiguration reloads configuration from the secret, and cnsOperator
+//with the latest configInfo
+func (volumeMigration *volumeMigration) reloadConfiguration(ctx context.Context) error {
+	log := logger.GetLogger(ctx)
+	cfg, err := common.GetConfig(ctx)
+	if err != nil {
+		log.Errorf("Failed to read config. Error: %+v", err)
+		return err
+	}
+	volumeMigrationInstance.cnsConfig = cfg
```

Comment:
We have a volume manager holding a vCenter connection. I guess that will updated by metadata syncer during reloadconfig, as we have supplied pointer. Can you double check this.